### PR TITLE
feat: Add more features on how attention component is positioned

### PR DESCRIPTION
**Pull request by felicia-haggqvist**
**Part of fixes for Jira-tickets:** [WARP-432](https://nmp-jira.atlassian.net/browse/WARP-432), [WARP-341](https://nmp-jira.atlassian.net/browse/WARP-341) & [WARP-373](https://nmp-jira.atlassian.net/browse/WARP-373)

- Add `distance`, `skidding`, `flip` and `fallbackPlacements` props to Attention component
- Add type-check for `fallbackPlacements` in the `connectedCallback()`
- Update `pointingAtDirection` to also account for the new placements: `top-start`, `top-end`, `right-start`, `right-end`, `left-start`, `left-end`, `bottom-start`, `bottom-end`
- Update `this._arrowDirectionClass` to also handle when placement includes either `-start` or `-end` 

I was unable to use `autoUpdatePosition` since it was causing an infinite update-loop (suspecting that it has something to do with where Lit allows you to add and remove event-listeners and the `autoUpdatePosition` was being used in the `updated` lifecycle method). So I had to manually add and remove event-listeners:
- Add event-listeners to the `connectedCallback()` for on `click`, `scroll`, `resize`, `touch`, `mouseover` and on `mouseout` depending on if Attention component is `tooltip`, `callout`, `popover` or `highlight`, and calls `handleDone`
- Add `handleDone` function that calls `recompute` (imported from the `warp-ds/core`-repo) and is wrapped in a window.requestAnimationFrame (helps to create a smoother and more efficient animations on web pages. See more info [here](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame)). 
- Calculation of arrowRotation has been removed from this repo and is now instead calculated in the `warp-ds/core` repo.

- Update Attention examples in `attention.html` to show how `distance`, `skidding`, `flip`, `fallbackPlacements` and the new placements can be used. 

- Bump `@warp-ds/core` to latest version to get recent changes

### CODE REVIEW

**Review by imprashast — APPROVED** [review]
Looks good. Not sure if this is still in Draft mode?

**Comment by felicia-haggqvist** [comment]
Nice! I've put this PR in draft mode because it still needs updates from the @warp-ds/css and @warp-ds/core dependencies. :)

File: pages/components/attention.html [hunk 1]
```diff
@@ -79,63 +79,135 @@ <h2 class="mt-24 mb-16">Props</h2>
...
+            <tr>
+              <td>flip</td>
+              <td>
+                <div>boolean</div>
+                <div class="annotation">Whether Attention element should flip its placement in order to keep it in view.</div>
+              </td>
+              <td>false</td>
+            </tr>
+            <tr>
+              <td>fallbackPlacements</td>
```
Comment (BalbinaK) [inline pages/components/attention.html:115]:
```suggestion
              <td>fallback-placements</td>
```

Comment (felicia-haggqvist) [reply pages/components/attention.html]: Updated in this commit:[150b818]( https://github.com/warp-ds/elements/pull/135/commits/150b8180215bfa0fac4cf9f3983db2c23d8650a7)

**Review by BalbinaK — APPROVED** [review]
LGTM provided it works SSR 😅

**Pull request merged**